### PR TITLE
Implement producer for RSQL SortInfo

### PR DESCRIPTION
**Pull request by janinko**

### CODE REVIEW

**Comment by janinko** [comment]
retest this please

File: facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/UniversalRSQLMapper.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    public <X extends GenericEntity<Integer>> Path<?> toPath(Class<X> type, From<?, X> from, RSQLSelectorPath selector){
+        return mapper(type).toPath(from, selector);
+    }
+
+    public <X extends GenericEntity<Integer>> String toPath(Class<X> type, RSQLSelectorPath selector){
+        return mapper(type).toPath(selector);
+    }
+
+    private <X extends GenericEntity<Integer>> RSQLMapper<X> mapper(Class<X> type){
+        for (RSQLMapper<?> mappers : mappers) {
```
Comment (michalovjan) [inline facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/UniversalRSQLMapper.java]: mappers ---> mapper

File: facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/UniversalRSQLMapper.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+@ApplicationScoped
+public class UniversalRSQLMapper {
+
+    @Inject
+    private Instance<RSQLMapper<?>> mappers;
+
+    public <X extends GenericEntity<Integer>> Path<?> toPath(Class<X> type, From<?, X> from, RSQLSelectorPath selector){
```
Comment (michalovjan) [inline facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/UniversalRSQLMapper.java]: I suggest renaming X generic to DB for readability

File: facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/BuildConfigurationRSQLMapper.java [hunk 3]
```diff
@@ -17,57 +17,54 @@
...
+            case "description": return BuildConfiguration_.description;
+            case "buildScript": return BuildConfiguration_.buildScript;
+            case "scmRevision": return BuildConfiguration_.scmRevision;
+            case "creationTime": return BuildConfiguration_.creationTime;
+            case "modificationTime": return BuildConfiguration_.lastModificationTime;
+            case "buildType": return BuildConfiguration_.buildType;
+            default: return null;
         }
     }
+/*
```
Comment (michalovjan) [inline facade/src/main/java/org/jboss/pnc/facade/rsql/mapper/BuildConfigurationRSQLMapper.java]: dlt pls!

File: facade/src/main/java/org/jboss/pnc/facade/rsql/AbstractTransformer.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+
+import javax.persistence.criteria.Path;
+import javax.persistence.criteria.Root;
+
+
+import javax.persistence.criteria.From;
+
+
+abstract class AbstractTransformer {
```
Comment (michalovjan) [inline facade/src/main/java/org/jboss/pnc/facade/rsql/AbstractTransformer.java]: not used currently, what is it doing here?

File: facade/src/main/java/org/jboss/pnc/facade/rsql/RSQLProducer.java [hunk 5]
```diff
@@ -19,28 +19,24 @@
...
      * map RSQL selector to Criteria API Path. The predicate can throw a runtime exception when used
      * if the query is bad.
      * @param <T> Type of the entity.
-     * @param toPath Function that converts RSQL selector to CriteriaAPI Path.
+     * @param type Entity type.
      * @param rsql The query RSQL.
      * @return Predicate representing the RSQL query.
      */
-    <T extends GenericEntity<Integer>> Predicate<T> getCriteriaPredicate(BiFunction<From<?, T>, RSQLSelectorPath, Path> toPath, String rsql);
+    <T extends GenericEntity<Integer>> Predicate<T> getCriteriaPredicate(Class<T> type, String rsql);
```
Comment (michalovjan) [inline facade/src/main/java/org/jboss/pnc/facade/rsql/RSQLProducer.java]: I suggest renaming T generic to DB for readability and consistency.

**Pull request merged**